dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -89,15 +89,36 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
     {
         if (IsPackagedAppScenario())
         {
-            return m_channel.PushNotificationReceived([weak_self = get_weak(), handler](auto&&, auto&& args)
+            if (m_channel)
             {
-                if (auto strong = weak_self.get())
+                return m_channel.PushNotificationReceived([weak_self = get_weak(), handler](auto&&, auto&& args)
                 {
-                    auto pushArgs = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(args);
-                    pushArgs.Handled(true);
-                    handler(*strong, pushArgs);
-                };
-            });
+                    if (auto strong = weak_self.get())
+                    {
+                        auto pushArgs = winrt::make<winrt::Microsoft::Windows::PushNotifications::implementation::PushNotificationReceivedEventArgs>(args);
+                        pushArgs.Handled(true);
+                        handler(*strong, pushArgs);
+                    };
+                });
+            }
+            else
+            {
+                auto lock = m_lock.lock_exclusive();
+
+                THROW_HR_IF_NULL(E_UNEXPECTED, m_channelInfo.channelUri.c_str());
```

Comment:
This is incorrect. It should be:
THROW_HR_IF_NULL(E_UNEXPECTED, m_channelInfo);

You essentially want to return an error if channelInfo is not set, right? So use the winrt reference, not the fields in the reference.

Reply:
Yeah I am aware of this. I have taken a note. Will update the PR :)

Reply:
Actually m_channelInfo is a struct static allocation object and not a pointer object. So checking on the channelUri string should be the right thing to do?

Reply:
Check out std::variant here. https://en.cppreference.com/w/cpp/utility/variant. You could technically use this and check if it is invalid or not?

Reply:
BTW c_str() can never be null for a string. It can only be empty.